File: src/main/java/com/oz/office_tastezip/global/exception/DataExistsException.java
package com.oz.office_tastezip.global.exception;

import com.oz.office_tastezip.global.response.ResponseCode;

import static com.oz.office_tastezip.global.response.ResponseCode.DUPLICATED_EMAIL;

public class DataExistsException extends GlobalException {

    public DataExistsException() {
        super(DUPLICATED_EMAIL);
    }

    public DataExistsException(String message) {
        super(DUPLICATED_EMAIL, message);
    }

    public DataExistsException(ResponseCode responseCode, String message) {
        super(responseCode, message);
    }
}

File: src/main/java/com/oz/office_tastezip/global/exception/DataExistsException.kt
package com.oz.office_tastezip.global.exception

import com.oz.office_tastezip.global.response.ResponseCode

class DataExistsException : GlobalException {
    constructor() : super(ResponseCode.DATA_EXIST)
    constructor(message: String) : super(ResponseCode.DATA_EXIST, message)
    constructor(responseCode: ResponseCode) : super(responseCode)
    constructor(responseCode: ResponseCode, message: String) : super(responseCode, message)
}
